### ISSUE TITLE
TextFields can't be interacted with when semantics are enabled

### ISSUE THREAD

**Issue by tugorez** [flutter/flutter#148285]
### Steps to reproduce

1. Go to  https://fwa11y.web.app/#/text-field-password or any other app with a text field using an Android or iOS device
2. Use a screen reader and enable semantics in the app
3. Try to type text and you'll see the erratic behavior.

### Expected results

Users should be able to easily type text into TextFields.

### Actual results

It seems very complicated to type text into textfields when a screen reader is enabled.

### Code sample

<details open><summary>Another sample app</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(App());
}

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Home(title: 'Flutter Focus'),
    );
  }
}

class Home extends StatefulWidget {
  const Home({super.key, required this.title});

  final String title;

  @override
  State<Home> createState() => _HomeState();
}

class _HomeState extends State<Home> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: SingleChildScrollView(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              for (int i = 0; i < 6; i++)
                if (i % 2 != 0)
                  const TextField(
                    decoration: InputDecoration(
                      border: OutlineInputBorder(),
                      labelText: 'Type Something',
                    ),
                  )
                else
                  TextButton(
                    onPressed: _incrementCounter,
                    child: const Text('Press me to increase count'),
                  ),
              Text('You have pushed the button this many times: $_counter'),
              const SizedBox(height: 100),
            ],
          ),
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

I'm able to reproduce this  since version 3.19.0 on both Android and iOS using talback and voiceover respectively.

**Comment by darshankawar** [comment]
Thanks for the report. I ran the given code sample using latest stable (3.22.0) on Android as well as on iOS and below is my observation:

1. Android (S10, running OS 12): With talkback ON and trying to type in the textfield, the talkback doesn't read the letters smoothly and stutters.
2. iOS (iphone 6s, running OS 15.3.1) : With voiceover ON and trying to type in the textfield, the talkback reads the letters smoothly as expected and didn't observe any jitters while doing so.

You mentioned that the behavior is same on Android and iOS. Can you provide short videos of both and sound enabled so that we can verify this further ?

**Comment by yjbanov** [comment]
Starting to look into this one now.

**Comment by yjbanov** [comment]
The culprit seems to be this change: https://github.com/flutter/engine/pull/43279. That change replaced `tap` with `didGainAccessibilityFocus`. However, `didGainAccessibilityFocus` is not sufficient to start an editing session in the framework. The result is that the browser focuses on the `<input>` element, shows the virtual keyboard, and allows the user to change the value in the text field. All this happens while the framework is entirely unaware of any editing, as there's no `TextInputConnection` established between the framework and the engine.

However, simply adding `tap` back to the "focus" handler will cause https://github.com/flutter/flutter/issues/128709 again. This situation ties back to https://github.com/flutter/flutter/issues/83809 where `didLoseAccessibilityFocus` does not result in the framework to remove the keyboard focus from the respective widget. As a result, the framework keeps sending `hasFocus` set to `true` on semantic nodes that should not have focus any more, which in return causes the browser to scroll to the semantic node that the user is not focusing on.

**Comment by yjbanov** [comment]
We're very close to landing `SemanticsAction.focus`, so I'll attempt to fix this issue together with #83809 in https://github.com/flutter/engine/pull/53134.
